OpenRA.Mods.Common/Traits/Cloak.cs

```diff
@@ -167,16 +195,50 @@ void ITick.Tick(Actor self)
 					cloakedToken = self.GrantCondition(Info.CloakedCondition);
 
 				// Sounds shouldn't play if the actor starts cloaked
-				if (!(firstTick && Info.InitialDelay == 0) && !otherCloaks.Any(a => a.Cloaked))
+				if (!(firstTick && Info.InitialDelay == 0) && (otherCloaks == null || !otherCloaks.Any(a => a.Cloaked)))
+				{
+					var pos = self.CenterPosition;
 					Game.Sound.Play(SoundType.World, Info.CloakSound, self.CenterPosition);
+
+					Func<WPos> posfunc = () => self.CenterPosition + Info.EffectOffset;
+					if (!Info.EffectTrackActor)
+						posfunc = () => pos + Info.EffectOffset;
+
+					if (Info.EffectImage != null && Info.CloakEffectSequence != null)
+					{
+						var palette = Info.EffectPalette;
+						if (Info.EffectPaletteIsPlayerPalette)
+							palette += self.Owner.InternalName;
+
+						self.World.AddFrameEndTask(w => w.Add(new SpriteEffect(
+							posFunc: posfunc, facingFunc: () => WAngle.Zero, w, Info.EffectImage, Info.CloakEffectSequence, palette)));
```

Comment:
Naming the parameters should not be necessary, or is it?
```suggestion
							posfunc, () => WAngle.Zero, w, Info.EffectImage, Info.CloakEffectSequence, palette)));
```

Reply:
Should be all fixed here and drop the test case